packages/ember-metal/lib/events.js

```diff
@@ -126,11 +126,7 @@ export function removeListener(obj, eventName, target, method) {
     target = null;
   }
 
-  metaFor(obj).removeFromListeners(eventName, target, method, (...args) => {
-    if ('function' === typeof obj.didRemoveListener) {
```

Comment:
callback is checked for being `function` inside `meta. removeFromListeners ` already